[Fix #391] get_platform

superberny70 opened the pull request:
Tal y como explique en su dia en la #391 he sustituido las constantes por una funciona q retorna el valor correcto.

SeiTaN commented:
@superberny70 la variable que has quitado `PLATFORM_NAME` tb se usa en api.py, por lo que va a fallar.

superberny70 commented:
gracias @SeiTaN lo corrigo y vuelvo a subir

superberny70 commented:
```
def get_platform(full_version=False):
    """
        Devuelve la información la version de xbmc o kodi sobre el que se ejecuta el plugin

        @param full_version: indica si queremos toda la informacion o no
        @type full_version: bool
        @rtype: str o dict
        @return: Si el paramentro full_version es True se retorna un diccionario con las siguientes claves:
            'num_version': (float) numero de version en formato XX.X
            'name_version': (str) nombre clave de cada version
            'plaform': (str) esta compuesto por "kodi-" o "xbmc-" mas el nombre de la version segun corresponda.
        Si el parametro full_version es False (por defecto) se retorna el valor de la clave 'plaform' del diccionario anterior.
        """
```

¿q os parece?

divadres commented:
A mi me parece bien, pero lo de full_version donde se va a usar? porque hay q mirar si hay q introducirlo en otras plataformas para que no de error, como mediaserver y en caso de introducirlo ¿Qué valores tendría que devolver?

SeiTaN commented on python/main-classic/core/config.py:
```diff
@@ -36,15 +36,37 @@
     num_version = xbmc.getInfoLabel('System.BuildVersion').split(' ')[0]
-    if num_version < 14:
-        return "xbmc-" + codename.get(num_version.split('.')[0], num_version)
+    ret['name_version'] = codename.get(num_version.split('.')[0], num_version)
```
Tengo una duda con esto.
¿Y ponerlo así?
`ret['name_version'] = codename.get(num_version.split('.')[0], "unknown_name_version%s" % num_version)`

Así si no encuentra la versión 20 por ejemplo (aunque no creo que se dé el caso nunca xDD) escribirá "unknown_name_version_20" en vez de "20" en la variable "name_version"

superberny70 replied:
Como bien dices no creo q se de el caso.  Acaba de salir la 17 y esto ya cubre la 18 también. 
Cuando se vayan presentando nuevas versiones,  si son compatibles ya se añadiran

superberny70 commented:
@divadres pensaba solo en kodi,  ya q en otras plataformas no creo q tenga demasiado sentido,  pero si es por un tema de compatibilidad se puede hacer q retorne un dict con los mismos claves y el valor q querais. 

Ahora no recuerdo exactamente en q punto necesite averiguar el numero de version de kodi... Puede ser en la biblioteca de kodi para averiguar el nombre de la bbdd?  Cosa q tb podria devolver en el mismo dict. 

Pero ya te digo q no esta pensado para canales o partes comunes,  mas bien solo para kodi.

divadres commented:
ok, si es así no hay problema, pero hay que tener en cuenta cuando se use que solo se puede usar si estamos en xbmc

superberny70 commented:
Bueno pues al final le voy a poner a todas las plataformas el full_version, aunq no sirva para nada por lo menos evitamos accidentes si alguien lo usa en algun canal.
Por otra parte mirare donde se esta usando xbmc.getInfoLabel('System.BuildVersion') y lo sustituyo por config.get_platform(True)

SeiTaN commented:
Actualice a la 17RC1 y me da un error
```
11:16:38.731 T:11664   ERROR:     [pelisalacarta.platformcode.launcher.run] Traceback (most recent call last):
                                              File "C:\Users\SeiTaN\AppData\Roaming\Kodi\addons\plugin.video.pelisalacarta\platformcode\launcher.py", line 109, in run
                                                itemlist = channelselector.getmainlist()
                                              File "C:\Users\SeiTaN\AppData\Roaming\Kodi\addons\plugin.video.pelisalacarta\channelselector.py", line 89, in getmainlist
                                                if "xbmceden" in config.get_platform():
                                              File "C:\Users\SeiTaN\AppData\Roaming\Kodi\addons\plugin.video.pelisalacarta\core\config.py", line 65, in get_platform
                                                ret['num_version'] = float (num_version)
                                            ValueError: invalid literal for float(): 17.1-RC1
```

divadres commented:
habrá que hacer un .split("-")[0], porque las betas y RC van con el guioncito....

superberny70 commented:
mejor usar expresiones regulares no creeis...
```
...
    num_version = xbmc.getInfoLabel('System.BuildVersion').split(' ')[0]
    num_version = re.match("\d+\.\d+",num_version).group(0)
    ret['name_version'] = codename.get(num_version.split('.')[0], num_version)
...
```
En cuanto pueda subo un fix
